src/pushcollector/_impl/proxy.py

```diff
@@ -50,11 +51,48 @@ class CollectorProxy(object):
     def __init__(self, delegate):
         self._delegate = delegate
 
+    def _translate_pushitem(self, pushitem):
+        def _get_checksum():
+            if pushitem.md5sum and pushitem.sha256sum:
+                return {"md5": pushitem.md5sum, "sha256": pushitem.sha256sum}
+            else:
+                return None
+
+        pushitem = {
+            "filename": pushitem.name,
+            "state": pushitem.state,
+            "src": pushitem.src,
+            "dest": str(pushitem.dest) if pushitem.dest else None,
+            "checksums": _get_checksum(),
+            "origin": pushitem.origin,
+            "build": pushitem.build,
+            "signing_key": pushitem.signing_key,
+        }
+
+        return pushitem
+
+    def _is_pushitem_obj(self, pushitem):
+        # check if the object is an instance of PushItem or
+        # its subclass using inspect instead of instanceof
+        # as pushsource depends on pushcollector module which
+        # leads to cyclic dependency
+        for item_cls in inspect.getmro(type(pushitem)):
+            if item_cls.__name__ == "PushItem" and item_cls.__module__.startswith(
+                "pushsource"
+            ):
+                return True
+
+        return False
+
     def update_push_items(self, items):
+        pushitems = []
         for item in items:
+            if self._is_pushitem_obj(item):
```

Comment:
This is the part which I would suggest changing.  Actually how about writing it like this:

```suggestion
            item_dicts = self._translate_pushitem(item)
```

`_translate_pushitem` itself can have logic "if input is a dict, just return [item], else convert it to dicts of the expected format".